config/routes.rb

```diff
@@ -1,3 +1,19 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root to: 'pages#home'
+
+
+
+    resources :events do  sponsors
+                          companies
+                          workshops do users
+                          end
+                          end
+    "event/book", to: "event#book"
+    "event/sponsor" to: "event#sponsor"
+    "event/workshop/attend" to: "workshop/attend"
```

Comment:
Maybe change to something like this:

"event/:id/attend" to: "event#attend" as "attend_path", method: :get <-- to render attend form
"event/:id/attend"  to: "event#attend" as "attend_path, method: :post <-- to save the attend form